### PR TITLE
Fix extra row in timeperiod start column B

### PR DESCRIPTION
**Pull request by siddharth-krishna**
I think the "Remove empty rows" line wasn't always working, leading to an extra row of start years B:
```
    D     B     E     M  Year
0   1  2018  2018  2018  2018
1   1  2019  2019  2019  2019
2   1  2020  2020  2020  2020
3  20  2021  2040  2030  2030
4  20  2041  2060  2050  2050
5      2081
```

I also updated the computation of B to be a bit more pandas-y, which should be faster and more concise.

### CODE REVIEW

File: times_excel_reader.py [hunk 1]
```diff
@@ -483,19 +483,16 @@ def process_time_periods_table(table: EmbeddedXlTable):
...
-            df.loc[i, 'B'] = df.loc[i-1, 'B'] + df.loc[i-1, 'D']
-
+        # Start years = start year, then cumulative sum of period durations
+        df['B'] = (active_series.cumsum() + start_year).shift(1, fill_value=start_year)
         df['E'] = df.B + df.D - 1
         df['M'] = df.B + ((df.D - 1) // 2)
         df['Year'] = df.M
 
-        return replace(table, dataframe=df)
+        return replace(table, dataframe=df.astype(int))
```
Comment (siddharth-krishna) [inline times_excel_reader.py:495]: Also changed the type to int since all values in this table are years

**Review by samwebster — APPROVED** [review]

**Comment by siddharth-krishna** [comment]
@samwebster a suggestion for github PRs: changing the default merge strategy to "Squash and merge" (and selecting the option to automatically delete PR branches after merge) helps keep the main branch commit history nice and clean.

**Comment by samwebster** [comment]
Good point. I've disabled merge and rebase, leaving only squash, and turned on branch deletion

**Comment by siddharth-krishna** [comment]
Thanks!

**Pull request merged**